numhask/src/NumHask/Algebra/Abstract/Field.hs

```diff
@@ -154,34 +148,37 @@ instance QuotientField Double Integer where
 -- > zero / zero != nan
 --
 -- Note the tricky law that, although nan is assigned to zero/zero, they are never-the-less not equal. A committee decided this.
-class (Semifield a) =>
+class (IntegralDomain a) =>
```

Comment:
Sometime in the future we can think of introducing `UpperBounded` and `LowerBounded` classes, which will serve as constraints in classes like this one.

Reply:
Semilattices, essentially